src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -23,12 +24,18 @@ internal partial class Binder
         /// </summary>
         private BoundExpression BindDeconstructionAssignment(AssignmentExpressionSyntax node, DiagnosticBag diagnostics)
         {
-            // receiver for first Deconstruct step
-            var boundRHS = BindValue(node.Right, diagnostics, BindValueKind.RValue);
-
             SeparatedSyntaxList<ArgumentSyntax> arguments = ((TupleExpressionSyntax)node.Left).Arguments;
-            ImmutableArray<DeconstructionVariable> checkedVariables = BindDeconstructionVariables(arguments, diagnostics);
+            ArrayBuilder<DeconstructionVariable> checkedVariables = BindDeconstructionVariables(arguments, diagnostics);
```

Comment:
Thanks for pointing this. I think when I switched to `ArrayBuilder` I removed the freeing and did not put it back. I'll fix.